src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,214 @@
+package org.broadinstitute.consent.http.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.http.entity.ContentType;
+import org.apache.http.nio.entity.NStringEntity;
+import org.broadinstitute.consent.http.configurations.ElasticSearchConfiguration;
+import org.broadinstitute.consent.http.db.DacDAO;
+import org.broadinstitute.consent.http.db.DataAccessRequestDAO;
+import org.broadinstitute.consent.http.db.DatasetDAO;
+import org.broadinstitute.consent.http.db.UserDAO;
+import org.broadinstitute.consent.http.models.Dataset;
+import org.broadinstitute.consent.http.models.DatasetProperty;
+import org.broadinstitute.consent.http.models.Study;
+import org.broadinstitute.consent.http.models.StudyProperty;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.elastic_search.DatasetTerm;
+import org.broadinstitute.consent.http.models.elastic_search.StudyTerm;
+import org.broadinstitute.consent.http.models.elastic_search.UserTerm;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.broadinstitute.consent.http.util.gson.GsonUtil;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+
+public class ElasticSearchService implements ConsentLogger {
+
+  private final RestClient esClient;
+  private final ElasticSearchConfiguration esConfig;
+  private final DatasetDAO datasetDAO;
+  private final DataAccessRequestDAO dataAccessRequestDAO;
+  private final DacDAO dacDAO;
+  private final UserDAO userDAO;
+  private final UseRestrictionConverter useRestrictionConverter;
+
+  public ElasticSearchService(
+      RestClient esClient,
+      ElasticSearchConfiguration esConfig,
+      UseRestrictionConverter useRestrictionConverter,
+      DatasetDAO datasetDAO,
+      DataAccessRequestDAO dataAccessRequestDAO,
+      DacDAO dacDAO,
+      UserDAO userDAO) {
+    this.esClient = esClient;
+    this.esConfig = esConfig;
+    this.useRestrictionConverter = useRestrictionConverter;
+    this.datasetDAO = datasetDAO;
+    this.dataAccessRequestDAO = dataAccessRequestDAO;
+    this.dacDAO = dacDAO;
+    this.userDAO = userDAO;
+  }
+
+
+  private static final String bulkHeader = """
+      { "index": {"_type": "dataset", "_id": "%d"} }
+      """;
+
+  public Response indexDatasets(List<DatasetTerm> datasets) throws IOException {
+    List<String> bulkApiCall = new ArrayList<>();
+
+    datasets.forEach((dsTerm) -> {
+      bulkApiCall.add(bulkHeader.formatted(dsTerm.getDatasetId()));
+      bulkApiCall.add(GsonUtil.getInstance().toJson(dsTerm) + "\n");
```

Comment:
In testing, I hit this error when indexing all datasets. I suspect it is originating from this line.
```
{
  "message": "class com.google.gson.JsonArray cannot be cast to class java.lang.String (com.google.gson.JsonArray is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')",
  "code": 500
}
```